scripts/bootloader/zb_ota_sign.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+"""nrfutil command line tool."""
+import os
+import sys
+import click
+import time
+import logging
+import re
+sys.path.append(os.getcwd())
+
+from zb_ota.package import Package
+import spinel.util as util
+
+
+def int_as_text_to_int(value):
+    try:
+        if value[:2].lower() == '0x':
+            return int(value[2:], 16)
+        elif value[:1] == '0':
+            return int(value, 8)
+        return int(value, 10)
+    except ValueError:
+        raise print('%s is not a valid integer' % value)
+
+
+def pause():
+    while True:
+        try:
+            input()
+        except (KeyboardInterrupt, EOFError):
+            break
+
+class BasedIntOrNoneParamType(click.ParamType):
+    name = 'Integer'
+
+    def convert(self, value, param, ctx):
+        try:
+            if value.lower() == 'none':
+                return 'none'
+            return int_as_text_to_int(value)
+        except ValueError:
+            print('%s is not a valid integer' % value, param, ctx)
+
+BASED_INT_OR_NONE = BasedIntOrNoneParamType()
+
+class BasedIntParamType(BasedIntOrNoneParamType):
+    name = 'Integer'
+
+BASED_INT = BasedIntParamType()
+
+def convert_version_string_to_int(s):
+    """Convert from semver string "1.2.3", to integer 10203"""
+    numbers = s.split(".")
+    if len(numbers) != 3:
+        raise click.BadParameter("Must be on the format x.y.z", param_hint='application-version-string')
+    js = [10000, 100, 1]
+    return sum([js[i] * int(numbers[i]) for i in range(3)])
```

Comment:
Returned value can be same for different versions (for example 1.1.0 and 1.0.100). Is this method good enough? The max value of part of semver is 1024